### ISSUE TITLE
DEBUG: improve displaying complex data types 

### ISSUE THREAD

**Issue by akharche** [IntelPython/numba-dppy#329]
Consider improving support of complex data types. With “vanilla” Numba, debugger can only analyse the contents of simple types, such as integers and doubles. But with a numpy array, debugger output starts to be difficult for an average user. For an example, see https://numba.pydata.org/numba-doc/latest/user/troubleshoot.html#running-with-gdb-enabled, in particular how the debugger shows the variable “c” (which is a numpy array). We should aim to show such types as single objects, similar to how we do it with C++ class objects.

**Comment by stuartarchibald** [comment]
https://github.com/numba/numba/pull/7177 creates the debuginfo for the `DataModel`'s of most of Numba's builtin types. This makes it possible to find out types and debug into the data models as structures. Some helper functions need writing to perhaps make the printed representations closer to those available from the python `__repr__` of the boxed types.

**Comment by akharche** [comment]
AR: Need to check with [numba/numba#7177](numba/numba#7177)

**Comment by akharche** [comment]
Numba fix [numba/numba#7177](https://github.com/IntelPython/numba-dppy/issues/numba/numba#7177) improves displaying of complex data types like arrays. It makes it possible to access "data" in arrays. It is possible to get array values by command `x/10f b.data`.

```python
import numpy as np
import numba_dppy as dppy
import dpctl


@dppy.kernel(debug=True)
def data_parallel_sum(a, b, c):
    i = dppy.get_global_id(0)
    l1 = a[i] + 2.5
    l2 = b[i] * 0.3
    c[i] = l1 + l2


global_size = 10
N = global_size

a = np.array(np.random.random(N), dtype=np.float32)
b = np.array(np.random.random(N), dtype=np.float32)
c = np.ones_like(a)

with dpctl.device_context("opencl:gpu"):
    data_parallel_sum[global_size, dppy.DEFAULT_LOCAL_SIZE](a, b, c)
```

```bash
Thread 2.1 hit Breakpoint 1, with SIMD lanes [0-7], __main__::data_parallel_sum () at sum_local_vars.py:22
22          i = dppy.get_global_id(0)
(gdb) info locals
a = {meminfo = 0x0, parent = 0x0, nitems = 0, itemsize = 0, data = 0x0, shape = {0}, strides = {0}}
b = {meminfo = 0x0, parent = 0x0, nitems = 0, itemsize = 0, data = 0x0, shape = {0}, strides = {0}}
c = {meminfo = 0x0, parent = 0x0, nitems = 0, itemsize = 0, data = 0x0, shape = {0}, strides = {0}}
i = 0
l1 = 0
l2 = 0
__ocl_dbg_gid0 = 0
__ocl_dbg_gid1 = 0
__ocl_dbg_gid2 = 0
__ocl_dbg_lid0 = 0
__ocl_dbg_lid1 = 0
__ocl_dbg_lid2 = 0
__ocl_dbg_grid0 = 0
__ocl_dbg_grid1 = 0
__ocl_dbg_grid2 = 0

(gdb) whatis a
type = array(float32, 1d, C) ({float addrspace(1)*, float addrspace(1)*, i64, i64, float addrspace(1)*, [1 x i64], [1 x i64]})
Thread 2.2 hit Breakpoint 1, with SIMD lanes [0-1], __main__::data_parallel_sum () at sum_local_vars.py:22
22          i = dppy.get_global_id(0)
(gdb) n
23          l1 = a[i] + 2.5
(gdb) info locals
a = {meminfo = 0x0, parent = 0x0, nitems = 10, itemsize = 4, data = 0x555559e0d000, shape = {10}, strides = {4}}
b = {meminfo = 0x0, parent = 0x0, nitems = 10, itemsize = 4, data = 0x5555599ea000, shape = {10}, strides = {4}}
c = {meminfo = 0x0, parent = 0x0, nitems = 10, itemsize = 4, data = 0x555559d8d000, shape = {10}, strides = {4}}
i = 8
l1 = 0
l2 = 0
__ocl_dbg_gid0 = 0
__ocl_dbg_gid1 = 17179869188
__ocl_dbg_gid2 = 17179869188
__ocl_dbg_lid0 = 17179869188
__ocl_dbg_lid1 = 8
__ocl_dbg_lid2 = 42949672970
__ocl_dbg_grid0 = 0
__ocl_dbg_grid1 = 10
__ocl_dbg_grid2 = 0
(gdb) n
24          l2 = b[i] * 0.3
(gdb) info locals
a = {meminfo = 0x0, parent = 0x0, nitems = 0, itemsize = 0, data = 0x0, shape = {0}, strides = {0}}
b = {meminfo = 0x0, parent = 0x0, nitems = 10, itemsize = 4, data = 0x5555599ea000, shape = {10}, strides = {4}}
c = {meminfo = 0x0, parent = 0x0, nitems = 10, itemsize = 4, data = 0x555559d8d000, shape = {10}, strides = {4}}
i = 8
l1 = 3.0279030203819275
l2 = 0
__ocl_dbg_gid0 = 0
__ocl_dbg_gid1 = 0
__ocl_dbg_gid2 = 0
__ocl_dbg_lid0 = 42949672970
__ocl_dbg_lid1 = 0
__ocl_dbg_lid2 = 93825068486656
__ocl_dbg_grid0 = 4612811918334230528
__ocl_dbg_grid1 = 0
__ocl_dbg_grid2 = 0
(gdb) x/10f b.data
0x5555597dc000: 0.950231791     0.422228962     0.782108009     0.252930403
0x5555597dc010: 0.846446514     0.994258225     0.226994261     0.144751802
0x5555597dc020: 0.186860502     0.856326342

```
